react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      username: '',
+      password: ''
+    };
+
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({ username: event.target.value });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  handleSubmit(event) {
+    console.log('state:', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <>
+        <form onSubmit={this.handleSubmit}>
+          <label>
+            Username
+          </label>
+          <input type="text" value={this.state.username} onChange={this.handleUsernameChange} />
+          <label>
+            Password
+          </label>
+          <input type="password" value={this.state.password} onChange={this.handlePasswordChange} />
+          <button formAction='submit'>Sign Up</button>
```

Comment:
Here you would use the `type` prop set to `submit`, but the default `type` is `submit` so it isn't required.

Reply:
Roger